components/script/script_runtime.rs

```diff
@@ -248,6 +250,35 @@ unsafe extern "C" fn empty(extra: *const c_void) -> bool {
     result
 }
 
+#[allow(unsafe_code)]
+unsafe extern "C" fn push_new_interrupt_queue(interrupt_queues: *mut c_void) -> *mut c_void {
+    let mut result = std::ptr::null_mut();
+    wrap_panic(&mut || {
+        let interrupt_queues = &mut *(interrupt_queues as *mut Vec<Rc<MicrotaskQueue>>);
+        let mut new_queue = Rc::new(MicrotaskQueue::default());
+        // TODO: SAFETY: is this ok? It subverts the Rc::get_mut() check
+        result = Rc::get_mut(&mut new_queue).expect("Guaranteed by usage of `new_queue`") as *mut _
+            as *mut c_void;
+        interrupt_queues.push(new_queue.clone());
+    });
+    result
+}
+
+#[allow(unsafe_code)]
+unsafe extern "C" fn pop_interrupt_queue(interrupt_queues: *mut c_void) -> *mut c_void {
+    let mut result = std::ptr::null_mut();
+    wrap_panic(&mut || {
+        let interrupt_queues = &mut *(interrupt_queues as *mut Vec<Rc<MicrotaskQueue>>);
+        let mut popped_queue: Rc<MicrotaskQueue> =
+            interrupt_queues.pop().expect("Guaranteed by SpiderMonkey?");
+        // Dangling, but jsglue.cpp will only use this for pointer comparison.
+        // TODO: SAFETY: is this ok? It subverts the Rc::get_mut() check
+        result = Rc::get_mut(&mut popped_queue).expect("Guaranteed by usage of `popped_queue`")
```

Comment:
Same as above.

Reply:
fixed in cacdb8f39f73c7260e825a41d0fb172f20fb13bd